react/Carousel.tsx

```diff
@@ -41,228 +38,33 @@ interface ArrowContainerProps {
-                  'ui:widget': 'image-uploader',
-                },
-              },
-              description: {
-                default: '',
-                title: 'admin/editor.carousel.banner.description.title',
-                type: 'string',
-              },
-              externalRoute: {
-                default: false,
-                isLayout: false,
-                title: 'admin/editor.carousel.banner.externalRoute.title',
-                type: 'boolean',
-              },
-            },
-            dependencies: {
-              externalRoute: {
-                oneOf: [
-                  {
-                    properties: {
-                      externalRoute: {
-                        enum: [false],
-                      },
-                      ...internalRouteSchema,
-                    },
-                  },
-                  {
-                    properties: {
-                      externalRoute: {
-                        enum: [true],
-                      },
-                      ...externalRouteSchema,
-                    },
-                  },
-                ],
-              },
-            },
-            title: 'admin/editor.carousel.banner.title',
-            type: 'object',
-          },
-          minItems: 1,
-          title: 'admin/editor.carousel.banners.title',
-          type: 'array',
-        },
-        height: {
-          default: 420,
-          enum: [420, 440],
-          isLayout: true,
-          title: 'admin/editor.carousel.height.title',
-          type: 'number',
-        },
-        showArrows: {
-          default: true,
-          isLayout: true,
-          title: 'admin/editor.carousel.showArrows.title',
-          type: 'boolean',
-        },
-        showDots: {
-          default: true,
-          isLayout: true,
-          title: 'admin/editor.carousel.showDots.title',
-          type: 'boolean',
-        },
-        autoplay: {
-          default: true,
-          isLayout: false,
-          title: 'admin/editor.carousel.autoplay.title',
-          type: 'boolean',
-        },
-      },
-      title: 'admin/editor.carousel.title',
-      type: 'object',
-    }
-    // tslint:enable
-  }
-
-  public state = {
-    currentSlide: 0,
-  }
-
-  public perPage = 1
+const PER_PAGE = 1
+const CSS_HANDLES = [
+  'arrow',
+  'arrowLeft',
+  'arrowRight',
+  'arrowsContainerWrapper',
+  'sliderRoot',
+  'sliderFrame',
+  'slide',
+  'activeDot',
+  'notActiveDot',
+  'containerDots',
+] as const
 
-  public handleChangeSlide = (i: number): void => {
-    this.setState({ currentSlide: i })
-  }
-
-  public handleNextSlide = (): void => {
-    this.setState(({ currentSlide }) => {
-      const bannersLength: number = this.props.banners.filter(
-        banner => banner && (banner.mobileImage || banner.image)
-      ).length
-      const nextSlide: number =
-        ((currentSlide + 1 - this.perPage) % bannersLength) + this.perPage
+function Carousel(props: Props) {
```

Comment:
Why not an arrow function?

Reply:
I usually prefer to use just functions because it seems cleaner and I'm not using `this` inside of it, but I gonna switch to arrow function